### PR TITLE
Migration to the official GIIS repository of the replication package

### PR DESCRIPTION
**Pull request by augustocristian**
This PR request contains:
- Necessary Java code to execute the experimentation
- In `/docs` folder with:
  - All the generated scenarios as well as their coverage matrix of the user requirements.
  - The test cases generated.
  - The modifications performed in the generated test cases to make them pass.
  - The summary of the results of each RQ
  - The experimental set-up explanation
- The README file that explains briefly the content of the RP
- The citation RF file that enables the "Cite us" button in github with the correct format

### CODE REVIEW

**Comment by augustocristian** [comment]
This is the first version of the RP in the official repository, two doubts that raise to me during the migration:
- Its necessary here sonar-actions?
- May be preferable convert the java multimodule project into a simple java project?
Thanks in advance

File: .gitignore
Comment (javiertuya) [inline .gitignore]: Simplificar gitignore con lo necesario

File: README.md
Comment (javiertuya) [inline README.md]: añadir link a zenodo ahora que está

File: README.md
Comment (javiertuya) [inline README.md:25]: maybe this should be after the replication package structure

File: README.md
Comment (javiertuya) [inline README.md]: Use v1.2.0 (I just created it as prerelease)
Use the term release instead of version to match the GitHub terms.

File: README.md
Comment (javiertuya) [inline README.md]: Although absolute paths may work, better to use relative paths. Check other places where you also use absolute paths

File: README.md
Comment (javiertuya) [inline README.md]: You should not be using non LTS java versions, in this case 21 should preferred

File: pom.xml
Comment (javiertuya) [inline pom.xml]: Do you need these plugins, if not, remove them

File: pom.xml
Comment (javiertuya) [inline pom.xml]: This repo will not generate artifacts for maven, remove these unnecessary parts.

File: retorch-llm-rp/.project
Comment (javiertuya) [inline retorch-llm-rp/.project]: This is still the template name

File: pom.xml
Comment (javiertuya) [inline pom.xml]: Change artifact id and name (e.g. to the name of this repo)

File: pom.xml
Comment (javiertuya) [inline pom.xml]: Do you plan to use THIS repo to other works, if so, it is ok to keep it as multimodule, if not, maybe this is not necessary

File: retorch-llm-rp/pom.xml
Comment (javiertuya) [inline retorch-llm-rp/pom.xml:17]: Either if this is finally a multimodule or not, check for removal all unnecessary stuff (dependencies and build steps) in the pom (or poms)

File: retorch-llm-rp/pom.xml
Comment (javiertuya) [inline retorch-llm-rp/pom.xml]: It seems a little bit inconsistent to use here 21, say that you compile with 22 and set 8 in other places. Just keep java 21 everywhere

File: retorch-llm-rp/src/test/java/giis/retorch/llmrp/TestClass1.java
Comment (javiertuya) [inline retorch-llm-rp/src/test/java/giis/retorch/llmrp/TestClass1.java]: This should be removed

File: retorch-llm-rp/src/test/resources/log4j2.xml
Comment (javiertuya) [inline retorch-llm-rp/src/test/resources/log4j2.xml]: Remove if you are not using loggers

File: pom.xml
Comment (javiertuya) [inline pom.xml]: Change the version number to 1.0.0
After merge, I will create a release with this number (as prerelease) to refer it from zenodo.

**Review by javiertuya — CHANGES_REQUESTED** [review]

File: retorch-llm-rp/src/test/resources/log4j2.xml
Comment (augustocristian) [reply retorch-llm-rp/src/test/resources/log4j2.xml]: I am using loggers in the experimentation classs

File: pom.xml
Comment (augustocristian) [reply pom.xml]: @javiertuya  I was thinking about rename the current project to retorch-llm-expstudy and leave the name of the repository retorch-llm-rp. What do you think?

File: README.md
Comment (augustocristian) [reply README.md]: I am waiting for @ClaudiodelaRiva  approval of the Zenodo draft, at the moment it would be available I update everything

**Comment by augustocristian** [comment]
I've made the necessary changes and created a new issue (https://github.com/giis-uniovi/retorch-llm-rp/issues/2) to address the restructuring of the project. I treated it as a separate task because doing it now would complicate the review of the current PR.

File: retorch-llm-rp/pom.xml
Comment (javiertuya) [reply retorch-llm-rp/pom.xml]: Project does not compile

File: pom.xml
Comment (javiertuya) [inline pom.xml:1]: Project does not compile

**Review by javiertuya — CHANGES_REQUESTED** [review]

File: pom.xml
Comment (augustocristian) [reply pom.xml:1]: I've cloned the repository in my laptop (no previous installation) and both (clean, install with maven work-compile). Also the intellij is able to compile and execute: 
image
 I am missing something? @javiertuya

Comment (augustocristian) [reply pom.xml:1]: I've also executed and works-generate the scenarios: 
image

Comment (javiertuya) [reply pom.xml:1]: @augustocristian As I have jdk 17, I changed the parent pom version, and it fails. I found that the module pom REDEFINES the java version, this should be removed

**Comment by augustocristian** [comment]
I've compared how the template project is configured and restored  the  initial configuration removing the unnecessary plugins. Sorry for the dependencies update, I've tested and the code remains working without this 3 dependencies updated

File: pom.xml
Comment (javiertuya) [inline pom.xml:44]: Still unnecessary dependencies and plugins (both poms)

**Review by javiertuya — CHANGES_REQUESTED** [review]

**Comment by augustocristian** [comment]
Sorry @javiertuya. 
I've removed the unused dependencies from early trials in the project as well as the plugin not used. The  maven-source-plugin is the only that is required to create the jars using maven, that remains in the father pom.xml

File: pom.xml
Comment (javiertuya) [inline pom.xml]: By removing the implementation you are losing the ability to generate logs. Also fix the log settings location

File: pom.xml
Comment (javiertuya) [inline pom.xml]: You do not need this to generate a jar that will not be published. Moreover, you do not need to generate a jar

**Review by javiertuya — CHANGES_REQUESTED** [review]

**Comment by augustocristian** [comment]
@javiertuya  is something wrong with the last changes? 
I've realized that  there's one unresolved conversation, its not the first time that  I mark something as resolved and appears later as not-resolved.

**Comment by javiertuya** [comment]
I marked it as unresolved, you didn't address the second part of the comment

File: pom.xml
Comment (augustocristian) [reply pom.xml]: I've completely lost the focus, the pom now is minimized to:
- Only the artifact, group, etc
- The dependencies required
- Our metadata
- Some properties like the maven compiler and target version and the encoding
- License
What can be simplified or removed without impacting the stability of the package manager? @javiertuya

Comment (augustocristian) [reply pom.xml]: Sorry! I've realized now about the <packaging> annotation, removed and pushed!

**Comment by javiertuya** [comment]
@augustocristian let's talk about this tomorrow during the group meeting

**Comment by augustocristian** [comment]
I've moved the log under /java/resources, now all is working

**Pull request merged**
